cli/daemon/apps/apps.go

```diff
@@ -0,0 +1,140 @@
+package apps
+
+import (
+	"database/sql"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/cockroachdb/errors"
+	"github.com/rs/zerolog/log"
+
+	"encr.dev/cli/daemon/internal/manifest"
+	"encr.dev/cli/internal/appfile"
+)
+
+var ErrNotFound = errors.New("app not found")
+
+// Instance describes an app instance known by the Encore daemon.
+type Instance struct {
+	root       string
+	localID    string
+	platformID string
+}
+
+func NewInstance(root, localID, platformID string) *Instance {
+	return &Instance{root, localID, platformID}
+}
+
+// Root returns the filesystem path for the app root.
+// It always returns a non-empty string.
+func (i *Instance) Root() string { return i.root }
+
+// LocalID reports a local, random id unique for this app,
+// as persisted in the .encore/manifest.json file.
+// It always returns a non-empty string.
+func (i *Instance) LocalID() string { return i.localID }
+
+// PlatformID reports the Encore Platform's ID for this app.
+// If the app is not linked it reports the empty string.
+func (i *Instance) PlatformID() string { return i.platformID }
+
+// PlatformOrLocalID reports PlatformID() if set and otherwise LocalID().
+func (i *Instance) PlatformOrLocalID() string {
+	if i.platformID != "" {
+		return i.platformID
+	}
+	return i.localID
+}
+
+func NewManager(db *sql.DB) *Manager {
+	return &Manager{db: db, instances: make(map[string]*Instance)}
+}
+
+// Manager keeps track of known apps and watches them for changes.
+type Manager struct {
```

Comment:
Could we add a list function to list all App's in the database?

I'm thinking the new autocomplete could then merge both the locally known instances with the platform's instances.

Reply:
Yeah that's a good idea, but I think we should hold off until we need it

Reply:
Well I already merged the autocomplete code, so I'd argue we kinda need it now ;)  - But fair can be a future PR

Reply:
Well what would it return beyond what you already have? The "LocalID" is not meaningful to users, it's just a randomly generated string, so not sure it should be an autocomplete option

Reply:
The slug for be able to generate a client for a running app using `--env=local` when it might not be linked